src/device.h

```diff
@@ -87,6 +87,8 @@ class device : public virtual device_interface, public info_container
 
     bool device_changed_notifications_on() const { return _device_changed_notifications; }
 
+    bool should_use_basic_formats() const;
```

Comment:
A few things: :)
- Not sure about the name - if we stuck to the existing convention, it would be `use_basic_formats_on()`?
- Should we really expose a function from the device? I'd bet the sensor pretty much knows how to get the context? If so, why not query the context directly and no go through the device? It's really not a device function, is it?

Reply:
The sensor uses it's device "owner" to access the context, no direct access.
Conceptually we can make this a per sensor decision, but I don't really see a case where it is not an application wide resolution. It is whether we convert at server side or host side and the CPU/bandwidth resources needed.

Reply:
I would argue it's actually per device, though. It's per context.
I think the sensor should query the context.

Reply:
Updated `sensor` to parse this parameter